broadbean/sequence.py

```diff
@@ -0,0 +1,1065 @@
+            delay: The required delay (s)
+
+        Raises:
+            ValueError: If a non-integer or non-non-negative channel number is
+                given.
+        """
+
+        self._awgspecs['channel{}_delay'.format(channel)] = delay
+
+    def setChannelFilterCompensation(self, channel: Union[str, int],
+                                     kind: str, order: int=1,
+                                     f_cut: float=None,
+                                     tau: float=None) -> None:
+        """
+        Specify a filter to compensate for.
+
+        The specified channel will get a compensation (pre-distorion) to
+        compensate for the specified frequency filter. Just to be clear:
+        the INVERSE transfer function of the one you specify is applied.
+        Only compensation for simple RC-circuit type high pass and low
+        pass is supported.
+
+        Args:
+            channel: The channel to apply this to.
+            kind: Either 'LP' or 'HP'
+            order: The order of the filter to compensate for.
+                May be negative. Default: 1.
+            f_cut: The cut_off frequency (Hz).
+            tau): The time constant (s). Note that
+                tau = 1/f_cut and that only one of the two can be specified.
+
+        Raises:
+            ValueError: If kind is not 'LP' or 'HP'
+            ValueError: If order is not an int.
+            SpecificationInconsistencyError: If both f_cut and tau are given.
+        """
+
+        if kind not in ['HP', 'LP']:
+            raise ValueError('Filter kind must either be "LP" (low pass) or '
+                             '"HP" (high pass).')
+        if not isinstance(order, int):
+            raise ValueError('Filter order must be an integer.')
+        if (f_cut is not None) and (tau is not None):
+            raise SpecificationInconsistencyError('Can not specify BOTH a time'
+                                                  ' constant and a cut-off '
+                                                  'frequency.')
+
+        keystr = 'channel{}_filtercompensation'.format(channel)
+        self._awgspecs[keystr] = {'kind': kind, 'order': order, 'f_cut': f_cut,
+                                  'tau': tau}
+
+    def addElement(self, position: int, element: Element) -> None:
+        """
+        Add an element to the sequence. Overwrites previous values.
+
+        Args:
+            position (int): The sequence position of the element (lowest: 1)
+            element (Element): An element instance
+
+        Raises:
+            ValueError: If the element has inconsistent durations
+        """
+
+        # Validation
+        element.validateDurations()
+
+        # make a new copy of the element
+        newelement = element.copy()
+
+        # Data mutation
+        self._data.update({position: newelement})
+
+        # insert default sequencing settings
+        self._sequencing[position] = {'twait': 0, 'nrep': 1,
```

Comment:
perhaps instead of putting default values here you can add optional arguments to "addElement", which can be used when creating this dictionary